fix 'React/RCTBridgeModule.h' file not found bug in hybrid ios app an…

icxcat opened the pull request:
Resolve RNOS.h: 'React/RCTBridgeModule.h' file not found bug in hybrid ios app and add podspec file to support react-native link by podfile# pod install way.

phillbaker commented:
Hi @icxcat, thanks for the PR.

To clarify, when you say "hybrid app", you have a parent swift/obj-c app that you're managing via Cocoapods and would like to include this package as well?

This header import should work in all versions of [react-native >= 0.40](https://tekkie.ro/quick-n-dirty/support-react-native-0.40-ios-headers/), is there a specific older version you're trying to support?

icxcat commented:
Podfile

# Uncomment the next line to define a global platform for your project
platform :ios, '10.0'
source 'https://github.com/CocoaPods/Specs.git'
target 'TimesWallet' do
  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!
  # Pods for TimesWallet
  pod 'React', :path => '../node_modules/react-native', :subspecs => [
  'Core',
  'CxxBridge',
  'DevSupport',
  'RCTActionSheet',
  'RCTAnimation',
  'RCTGeolocation',
  'RCTImage',
  'RCTLinkingIOS',
  'RCTNetwork',
  'RCTSettings',
  'RCTText',
  'RCTVibration',
  'RCTWebSocket'
  ]
  pod "yoga", :path => "../node_modules/react-native/ReactCommon/yoga"
  pod 'DoubleConversion', podspec: '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'Folly', podspec: '../node_modules/react-native/third-party-podspecs/Folly.podspec'
  pod 'glog', podspec: '../node_modules/react-native/third-party-podspecs/GLog.podspec'
  pod 'RNOS', :path => '../node_modules/react-native-os'
  pod 'react-native-randombytes', :path => '../node_modules/react-native-randombytes'
  pod 'RNSnackbar', :path => '../node_modules/react-native-snackbar'
  pod 'react-native-sqlite-storage', :path => '../node_modules/react-native-sqlite-storage'
  pod 'RNVectorIcons', :path => '../node_modules/react-native-vector-icons'
end

icxcat commented:
Podfile

pod 'React', :path => '../node_modules/react-native', :subspecs => [
  'Core',
  'CxxBridge',
  'DevSupport',
  'RCTActionSheet',
  'RCTAnimation',
  'RCTGeolocation',
  'RCTImage',
  'RCTLinkingIOS',
  'RCTNetwork',
  'RCTSettings',
  'RCTText',
  'RCTVibration',
  'RCTWebSocket'
  ]
  pod "yoga", :path => "../node_modules/react-native/ReactCommon/yoga"
  pod 'DoubleConversion', podspec: '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'Folly', podspec: '../node_modules/react-native/third-party-podspecs/Folly.podspec'
  pod 'glog', podspec: '../node_modules/react-native/third-party-podspecs/GLog.podspec'
  pod 'RNOS', :path => '../node_modules/react-native-os'
  pod 'react-native-randombytes', :path => '../node_modules/react-native-randombytes'
  pod 'RNSnackbar', :path => '../node_modules/react-native-snackbar'
  pod 'react-native-sqlite-storage', :path => '../node_modules/react-native-sqlite-storage'
  pod 'RNVectorIcons', :path => '../node_modules/react-native-vector-icons'
end

As you can see, I added the podspec and removed the duplicate references in the header file, and the project compiles and runs normally. Maybe this bug will only appear in hybrid development...
